templates/courses.html

```diff
@@ -60,6 +67,27 @@ <h2>List of Courses</h2>
     <script>
         $(document).ready(function () {
             handleYearSelectChange();
+
+            $('.badge-checkbox').change(function () {
+                changeBadgeColorCheckbox($(this));
+                filter("course");
+            });
+
+            // Check All button functionality
+            $('#checkAllBtn').click(function () {
+                $('.badge-checkbox').prop('checked', true).each(function () {
+                    changeBadgeColorCheckbox($(this));
+                });
+                filter("course");
+            });
+
+            // Uncheck All button functionality
+            $('#uncheckAllBtn').click(function () {
+                $('.badge-checkbox').prop('checked', false).each(function () {
+                    changeBadgeColorCheckbox($(this));
+                });
+                filter("course");
+            });
```

Comment:
This whole snippet could also go to `filter.html`. Those are event function definitions and do not need to run after the document is ready.